### PR TITLE
fr-process: fix memory leak

### PR DESCRIPTION
**Pull request by rbuj**
```
CFLAGS="-ggdb3 -O0 -fsanitize=address" LDFLAGS="-fsanitize=address" ./autogen.sh --prefix=/usr && make && sudo make install
```
```
Direct leak of 128 byte(s) in 1 object(s) allocated from:
    #0 0x7fec22178cb8 in __interceptor_realloc (/lib64/libasan.so.6+0xaecb8)
    #1 0x7fec210e02d7 in g_realloc ../glib/gmem.c:171
    #2 0x7fec2108da30 in g_ptr_array_maybe_expand ../glib/garray.c:1525
    #3 0x7fec2108e2d1 in g_ptr_array_add ../glib/garray.c:1834
    #4 0x47d3b9 in fr_process_begin_command /home/robert/builddir.gcc/engrampa/src/fr-process.c:362
    #5 0x465552 in begin_tar_command /home/robert/builddir.gcc/engrampa/src/fr-command-tar.c:228
    #6 0x4655f3 in fr_command_tar_list /home/robert/builddir.gcc/engrampa/src/fr-command-tar.c:239
    #7 0x447499 in fr_command_list /home/robert/builddir.gcc/engrampa/src/fr-command.c:612
    #8 0x43ac6f in load_local_archive /home/robert/builddir.gcc/engrampa/src/fr-archive.c:1146
    #9 0x43ad2f in copy_remote_file_done /home/robert/builddir.gcc/engrampa/src/fr-archive.c:1158
    #10 0x43ae4a in copy_remote_file_done_cb /home/robert/builddir.gcc/engrampa/src/fr-archive.c:1185
    #11 0x7fec210d1c3e in g_idle_dispatch ../glib/gmain.c:5897
    #12 0x7fec210d6e16 in g_main_dispatch ../glib/gmain.c:3381
    #13 0x7fec210d6c5f in g_main_context_dispatch ../glib/gmain.c:4099
    #14 0x7fec210d7181 in g_main_context_iterate ../glib/gmain.c:4175
    #15 0x7fec210d7202 in g_main_context_iteration ../glib/gmain.c:4240
    #16 0x7fec2136e4ce in g_application_run ../gio/gapplication.c:2569
    #17 0x4c61e0 in main /home/robert/builddir.gcc/engrampa/src/main.c:353
    #18 0x7fec20d5bb74 in __libc_start_main (/lib64/libc.so.6+0x27b74)
```

### CODE REVIEW

**Comment by raveit65** [comment]
@rbuj 
I tested your command for checking memleaks, but it doesn't work here.

**Comment by raveit65** [comment]
OK, `libasan` was missing in my installation. Now compiling with `-fsanitize=address` works.

File: src/fr-process.c [hunk 1]
```diff
@@ -330,9 +330,9 @@ fr_process_finalize (GObject *object)
 	process = FR_PROCESS (object);
 
 	fr_process_stop_priv (process, FALSE);
-	fr_process_clear (process);
 
-	g_ptr_array_free (process->priv->comm, FALSE);
+	g_ptr_array_foreach (process->priv->comm, (GFunc) fr_command_info_free, NULL);
```
Comment (raveit65) [inline src/fr-process.c:334]: This creates a new warning. https://app.travis-ci.com/github/mate-desktop/engrampa/jobs/557609821#L3388
```
fr-process.c:334:51: warning: cast between incompatible function types from 'void (*)(FrCommandInfo *)' to 'void (*)(void *, void *)' [-Wcast-function-type]

  334 |         g_ptr_array_foreach (process->priv->comm, (GFunc) fr_command_info_free, NULL);
´´´

Comment (rbuj) [reply src/fr-process.c]: done

**Comment by lukefromdc** [comment]
What is the status of this?

**Review by raveit65 — APPROVED** [review]
LGTM:
New Warning is gone and it runs well.

**Pull request merged**